programs/local/LocalServer.cpp

```diff
@@ -237,10 +456,97 @@ try
     registerDisks();
     registerFormats();
 
+    processConfig();
+    applyCmdSettings(global_context);
+
+    connect();
+
+    if (is_interactive)
+    {
+        std::cout << std::endl;
+        runInteractive();
+    }
+    else
+    {
+        runNonInteractive();
+
+        if (server_exception)
+            server_exception->rethrow();
+        if (client_exception)
+            client_exception->rethrow();
+    }
+
+    connection.reset();
+
+    global_context->shutdown();
+    global_context.reset();
+
+    status.reset();
+    cleanup();
+
+    return Application::EXIT_OK;
+}
+catch (const Exception & e)
+{
+    try
+    {
+        cleanup();
+    }
```

Comment:
Why do we reset connection, contexts and status before exiting only if no exception was thrown?